### ISSUE TITLE
Support some diagram language (written inline or in dedicated files)

### ISSUE THREAD

**Issue by kwin** [vsch/flexmark-java#528]
**Is your feature request related to a problem? Please describe.** 
Please support generating diagrams out of some diagram language like [mermaid](https://mermaid-js.github.io/mermaid/#/), [PlantUML](https://mermaid-js.github.io/mermaid/#/). Those could either be written inline or just referenced from markdown. That should lead to generating the SVG during build and afterwards generate HTML with an image tag referencing the generated SVG.

**Describe the solution you'd like**
Something similar to https://github.com/danielyaa5/puml-for-markdown but working in Java (locally)

**Describe alternatives you've considered**
The current solutions usually require NodeJS or another native binary (and are therefore not cross-platform)

**Comment by DaveJarvis** [comment]
Mermaid does not produce SVG files that conform to the SVG specification, which means that they do not render outside of a web browser environment. This is a [known issue](https://github.com/mermaid-js/mermaid/issues/2485). EchoSVG has taken [some strides](https://github.com/css4j/echosvg/issues/77) to handle rasterizing Mermaid's non-compliant output.

[KeenWrite](https://github.com/DaveJarvis/keenwrite) adds a [FencedBlockExtension](https://github.com/DaveJarvis/KeenWrite/blob/main/src/main/java/com/keenwrite/processors/markdown/extensions/fences/FencedBlockExtension.java#L37) to flexmark-java that converts of `diagram-` images to SVG diagrams via https://kroki.io/.

The advantage of the `diagram-` prefix is that it allows KeenWrite to automatically get new text-based diagram image formats without having to rebuild and redistribute the software.

Here's a screenshot showing a family tree rendered using `diagram-graphviz`:

![keenwrite-diagram](https://github.com/vsch/flexmark-java/assets/2131950/5294e930-ab05-4816-bb68-f8d933d99012)

That is, the Markdown document contains:

```
``` diagram-graphviz
digraph G2 {
  edge [dir=none, color="#`r#x(v$palette$body)`", fontcolor="#`r#x(v$palette$body)`"];
  node [shape=box, style="filled", margin="0.2, 0.225", fontname="Roboto", color="#`r#x(v$palette$body)`", fillcolor="#eef4d7"];
  graph [splines=ortho];
...
}
```

This supports all formats offered by Kroki: PlantUML, GraphViz, Mermaid (rendering issues due to non-compliance notwithstanding), etc. When the next version of EchoSVG is released, it should fix some of the Mermaid rendering issues.

**Comment by DaveJarvis** [comment]
> Any chance you can decouple that flexmark extension from the UI and provide that as dedicated artifact?

Good idea. I won't be taking this on, but feel free to add an issue to KeenWrite's issue tracker, where it might get some traction.

FWIW, KeenWrite can be run from the command-line without opening the UI, if that's your concern.

**Comment by clach04** [comment]
KeenWrite moved, command line doc link for above 404s, see https://gitlab.com/DaveJarvis/KeenWrite/-/blob/main/docs/cmd.md?ref_type=heads instead.
